### ISSUE TITLE
Push Notification not delivered to App on iOS if user opens app directly after receiving it

### ISSUE THREAD

**Issue by chrisweight** [urbanairship/phonegap-ua-push#225]
If I send a push notification to iOS devices with the app in the background, I see it is received on the device, but the app does not have the JSContext.

So in the scenario where - after receiving a notification we tap directly on the app icon, rather than the notification itself, the app will never be aware of the Push Notification, so we cannot locally store it (for instance to log what a user has received in a timeline)

On Android, the app still retains the jscontext and receives a push event, even if in the background.

Expected behaviour: on iOS, we are able to receive the same event as in Android to log / store a Push Notification when it is received and the user does _not_ tap directly on the notification.

The only way around this on iOS appears to be combining a Push Notification with an in-app message.

**Comment by rlepinski** [comment]
This is the difference between Android and iOS. On Android the app is started to receive and handle the message. On iOS, the OS post the notification on the apps behave without ever waking up the application. 

The only exception to this rule is if you sent a `content-available` push, the app will be woken up when the push is received on the device. However this is unreliable as Apple reserves the right to throttle content-available push and might decide to skip waking the app up. When you send an in-app message, its using the `content-available` flag. You can use that flag without an in-app message by setting the `"content_available": true` in the iOS content override if you using the push API, or turn on "background processing" in the delivery option if you are using the composer.

**Comment by chrisweight** [comment]
Hmm, background-processing consistently fails from the composer, in that case

**Comment by chrisweight** [comment]
As that's what we've been doing :(

**Comment by rlepinski** [comment]
You can verify if the flag is actually being set by looking at the received payload. If it has "content-available" in it then its working.

**Comment by rlepinski** [comment]
@chrisweight I just verified the composer flag is working fine for me. Make sure you have background app refresh enabled

**Comment by chrisweight** [comment]
Morning, ok - so I've also verified the `content-available` flag is coming through to device:

Running from XCode to ensure the device/app energy+data budget throttle is not applied, when receiving a PN, this is logged in the console:

```
2018-05-29 10:58:23.640539+0100 inspire[1789:524327] [I] +[UAAppIntegration handleIncomingNotification:foregroundPresentation:completionHandler:] [Line 230] Received notification: {
    "_" = "4a46a704-8cf7-4a87-bc7c-027020bcf46f";
    aps =     {
        alert =         {
            body = "So, content-available is there, but not actually doing anything";
            title = "Debugging iOS";
        };
        badge = 1;
        "content-available" = 1;
    };
    "com.urbanairship.metadata" = "eyJ2ZXJzaW9uX2lkIjoxLCJ0aW1lIjoxNTI3NTg3OTAzMjc4LCJwdXNoX2lkIjoiYTQzOTJhZGEtOWUzOS00OWFiLWI0MjktMWFjNWVkMzYxNzZlIiwiY2FtcGFpZ25zIjp7ImNhdGVnb3JpZXMiOlsidGVzdC1jYW1wYWlnbiJdfX0=";
}
2018-05-29 10:58:23.647619+0100 inspire[1789:524327] [D] +[UAActionRunner runActionWithName:value:situation:metadata:completionHandler:] [Line 60] No action found with name aps, skipping action.
2018-05-29 10:58:23.647949+0100 inspire[1789:524327] [D] +[UAActionRunner runActionWithName:value:situation:metadata:completionHandler:] [Line 60] No action found with name com.urbanairship.metadata, skipping action.
2018-05-29 10:58:23.649807+0100 inspire[1789:524327] [D] +[UAActionRunner runActionWithName:value:situation:metadata:completionHandler:] [Line 60] No action found with name _, skipping action.
```

However this is _not_ being made available to the Cordova app - No 'push' event is queued/received by the application to process when it returns to the foreground through the user tapping on the app icon (as opposed to tapping on the Push Notification).

I've also run on device outside of XCode and noted that the energy and data budgets are both well below the daily limit at this point.

It looks like we're also receiving the `title / message` stanzas at this point, though they're also not coming through to the app, which links to:

(#224)[https://github.com/urbanairship/urbanairship-cordova/issues/224]

**Comment by rlepinski** [comment]
It looks like the plugin is not registered to receive background push handling. Ill add a ticket to our backlog. One of our engineers is currently working on the plugin so he may be able to get to it for the next release.

**Comment by rlepinski** [comment]
@chrisweight Should be fixed in 7.1.0.
